C_image_processing/main.c

```diff
@@ -102,61 +125,81 @@ static void activate(GtkApplication *app, gpointer user_data)
     window = gtk_application_window_new(app);
     gtk_window_set_title(GTK_WINDOW(window), "C Image Processing");
     gtk_window_set_default_size(GTK_WINDOW(window), 1000, 1000);
-
     // create new vertical box container
     box = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
     gtk_widget_set_halign(box, GTK_ALIGN_CENTER);
     gtk_widget_set_valign(box, GTK_ALIGN_END);
-
     // include box container in window
     gtk_window_set_child(GTK_WINDOW(window), box);
 
-    // create new horizontal box container for image
-    image_box = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
-    gtk_widget_set_halign(image_box, GTK_ALIGN_CENTER);
-    gtk_widget_set_valign(image_box, GTK_ALIGN_CENTER);
-    gtk_widget_set_margin_bottom(image_box, 40);
-
-    // append image box to vertical box
-    gtk_box_append(GTK_BOX(box), image_box);
-
+    // create new horizontal box container for user image
+    user_image_box = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+    gtk_widget_set_halign(user_image_box, GTK_ALIGN_CENTER);
+    gtk_widget_set_valign(user_image_box, GTK_ALIGN_CENTER);
+    gtk_widget_set_margin_bottom(user_image_box, 40);
+    // append user image box to vertical box
+    gtk_box_append(GTK_BOX(box), user_image_box);
     // create image widget
     image = gtk_image_new_from_file("assets/images/test_image.bmp");
-    gtk_image_set_pixel_size(GTK_IMAGE(image), 900);
-
-    // append image widget to image box
-    gtk_box_append(GTK_BOX(image_box), image);
+    gtk_image_set_pixel_size(GTK_IMAGE(image), 450);
+    // append user image widget to user image box
+    gtk_box_append(GTK_BOX(user_image_box), image);
+
+    // create new horizontal box container for filtered image
+    filtered_image_box = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+    gtk_widget_set_halign(filtered_image_box, GTK_ALIGN_CENTER);
+    gtk_widget_set_valign(filtered_image_box, GTK_ALIGN_CENTER);
+    gtk_widget_set_margin_bottom(filtered_image_box, 40);
+    // append filtered image box to vertical box
+    gtk_box_append(GTK_BOX(box), filtered_image_box);
+    // create image widget
+    image = gtk_image_new_from_file("assets/images/test_image.bmp");
+    gtk_image_set_pixel_size(GTK_IMAGE(image), 450);
```

Comment:
Image on 1080 p if there are more than 1 blocks out the content below it!

Reply:
Fixed by reducing default size to 800x800 from 1000x1000